awx/api/serializers.py

```diff
@@ -4980,9 +4983,18 @@ def validate_node_state(self, value):
         return value
 
     def validate_hostname(self, value):
+        """
+        - Hostname cannot be "localhost" - but can be something like localhost.domain
+        - Cannot change the hostname of an-already instantiated & initialized Instance object
+        """
         if self.instance and self.instance.hostname != value:
             raise serializers.ValidationError("Cannot change hostname.")
 
+        HostnameRegexValidator(regex="[a-z0-9]([-a-z0-9]*[a-z0-9])?", flags=re.IGNORECASE)(value)
```

Comment:
I think for this validator we should override the `regex` and `flags` in `__init__()` instead of requiring them to be passed in.  If as I suspect the localhost checks can be done separately with a plain `RegexValidator`, we won't need the flexibility to swap out different regexes.